rust_src/src/process.rs

```diff
@@ -118,8 +118,8 @@ pub fn process_buffer(process: LispProcessRef) -> LispObject {
 /// This is the pid of the external process which PROCESS uses or talks to.
 /// For a network, serial, and pipe connections, this value is nil.
 #[lisp_fn]
-pub fn process_id(process: LispObject) -> Option<EmacsInt> {
-    let pid = unsafe { pget_pid(process.as_process_or_error().as_ptr()) };
```

Comment:
Will this signal the same error if the argument is not a process?

Reply:
Yes. The way it works is `as_foo_or_error` is called if `Foo` is used and `as_foo` is called when `Option<Foo>` is used. All of this is managed by supplying both `from(LispObject) -> Foo` and `from(LispObject) -> Option<Foo>`.